### PR TITLE
ruamel yaml lower bound to 0.17.10 and test

### PR DESCRIPTION
**Pull request by crazygao**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/setup.py [hunk 1]
```diff
@@ -45,7 +45,7 @@
     "strictyaml>=1.5.0,<2.0.0",  # used to identify exact location of validation error
     "waitress>=2.1.2,<3.0.0",  # used to serve local service
     "opencensus-ext-azure<2.0.0",  # configure opencensus to send telemetry to azure monitor
-    "ruamel.yaml>=0.17.35,<1.0.0",  # used to generate connection templates with preserved comments
+    "ruamel.yaml==0.17.10",  # used to generate connection templates with preserved comments
```
Comment (zhengfeiwang) [inline src/promptflow/setup.py:48]: why pin? I think we can use this as lower bound

Comment (zhengfeiwang) [inline src/promptflow/setup.py:48]: may add this to CHANGELOG

Comment (crazygao) [reply src/promptflow/setup.py:48]: pin to check the error if there are conflict in this list.

Comment (crazygao) [reply src/promptflow/setup.py:48]: Sure, of course

**Review by zhengfeiwang — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

File: src/promptflow/CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,9 @@
 ### Features Added
 - [SDK/CLI] Support `pfazure run cancel` to cancel a run on Azure AI.
 
+### Improvements
+- Bump `ruamel.yaml` lower bound to 0.17.10.
```
Comment (D-W-) [inline src/promptflow/CHANGELOG.md:9]: Change to lower bound.

**Pull request merged**